File: app/src/main/java/com/omkar/jet2demo/Jet2Application.java
package com.omkar.jet2demo;

import android.app.Application;

import com.omkar.jet2demo.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

public class Jet2Application extends DaggerApplication {


    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
         return DaggerAppComponent.builder().application(this).build();
    }
}

File: app/src/main/java/com/omkar/jet2demo/Jet2Application.kt
package com.omkar.jet2demo

import com.omkar.jet2demo.data.local.AppDatabase
import com.omkar.jet2demo.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import io.realm.Realm
import javax.inject.Inject

open class Jet2Application : DaggerApplication(), HasAndroidInjector {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.builder().application(this).build()
    }

    override fun onCreate() {
        super.onCreate()
        Realm.init(this)
    }

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Any>


    override fun androidInjector() = dispatchingAndroidInjector

    override fun onTerminate() {
        super.onTerminate()
        AppDatabase.instance?.closeDB()
    }

}